### DISCUSSION TITLE
Archive logs garbage collection

### DISCUSSION THREAD

**Discussion by grograxa** [argoproj/argo-workflows#13338] [Q&A] [answered]
When `archiveLogs` is set to true the main container logs are written to the artifact repository.  

Garbage collection when a workflow is deleted then takes two paths:

* If the workflow generated an artifact the artifact and logs are garbage collected and deleted from the repository. 
* If the workflow did not generate an artifact then garbage collection does not happen and the logs are left in the artifact repository. 

Is this the desired behaviour?

**Comment by agilgur5** [comment, marked as answer]
Someone filed a bug report about this in #13421, so marking this as superseded. They also found a workaround as well

> Is this the desired behaviour?

I don't know if that was intended or not; I imagine archive logs may not have been considered in the design of ArtifactGC and so that behavior may be indeterminate (cc @juliev0).

Since we don't recommend archive logs in [the docs](https://argo-workflows.readthedocs.io/en/release-3.5/configure-archive-logs/), I'm not sure it makes sense to fix or change this; we may very well remove the archive logs feature entirely

**Reply by juliev0** [reply]
You're right, it was not intended behavior.
